prereise/gather/hiflddata/data_trans.py

```diff
@@ -0,0 +1,569 @@
+        sub_name_dict[row['NAME']].append((row['LATITUDE'], row['LONGITUDE']))
+    return sub_by_coord_dict, sub_name_dict
+
+
+def Write_sub(clean_data, zone_dic):
+    """Write the data to sub.csv as output
+
+    :param pandas.DataFrame clean_data: substation dataframe as returned by :func:`Clean`
+    :param dict zone_dic: zone dict as returned by :func:`get_Zone`
+    """
+
+    with open('output/sub.csv','w',newline='') as sub:
+        csv_writer = csv.writer(sub)
+        csv_writer.writerow(["sub_id","sub_name","lat","lon","zone_id","type"])
+        for index, row in clean_data.iterrows():
+            csv_writer.writerow([row['ID'],row['NAME'],row['LATITUDE'],row['LONGITUDE'],zone_dic[row['STATE']],row['TYPE']])
+
+
+def Write_Bus(clean_data, zone_dic, KV_dict):
+    """Write the data to bus.csv as output
+
+    :param pandas.DataFrame clean_data: substation dataframe as returned by :func:`Clean`
+    :param dict zone_dic: zone dict as returned by :func:`get_Zone`
+    :param dict KV_dict: substation KV dict
+    """
+
+    with open('output/bus.csv', 'w', newline='') as bus:
+        csv_writer = csv.writer(bus)
+        csv_writer.writerow(["Bus_id", "PD", "Zone_id", "base_KV"])
+        missingSub = []
+        for index, row in clean_data.iterrows():
+            sub = (row['LATITUDE'], row['LONGITUDE'])
+            if (sub in KV_dict):
+                csv_writer.writerow([row['ID'], 0, zone_dic[row['STATE']], KV_dict[sub]])
+            else:
+                missingSub.append(sub)
+
+    print("INFO: ", len(missingSub), " substations excluded from the network. Some examples:")
+    print(missingSub[:20])
+
+
+def Write_bus2sub(clean_data):
+    """Write the data to bus2sub.csv as output
+
+    :param pandas.DataFrame clean_data: substation dataframe as returned by :func:`Clean`
+    """
+
+    with open('output/bus2sub.csv','w',newline='') as bus2sub:
+        csv_writer = csv.writer(bus2sub)
+        csv_writer.writerow(["Bus_id","sub_id"])
+        for index, row in clean_data.iterrows():
+            csv_writer.writerow([row['ID'],row['ID']])
+
+
+def Write_branch(lines):
+    """Write the data to branch.csv as output
+
+    :param list lines:  a list of lines as returned by :func:`Neighbors`
+    """
+
+    with open('output/branch.csv', 'w', newline='') as branch:
+        csv_writer = csv.writer(branch)
+        csv_writer.writerow(['branch_id', 'line_type', 'from_bus_id', 'from_bus_name', 'to_bus_id', 'to_bus_name', 'length_in_mile', 'reactance', 'rateA'])
+        csv_writer.writerows(lines)
+
+
+# Reactance Calculate 1
+kv_xperunit_calculate_1 = (
+    (69, 0.2293),
+    (115, 0.1106),
+    (230, 0.0567),
+    (345, 0.0074),
+    (500, 0.0001),
+    (1100, 0.0001)
+)
+
+# Reactance Calculate 2
+kv_xperunit_calculate_2 = (
+    (69, 0.0096),
+    (100, 0.0063),
+    (115, 0.0039),
+    (138, 0.0026),
+    (161, 0.0021),
+    (230, 0.0011),
+    (345, 0.0005),
+    (500, 0.0002),
+    (765, 0.0001),
```

Comment:
Can we increase the number of significant digits for some of these smaller numbers?

Reply:
These number are also from Yifan's doc. I guess we can discuss with Yifan again on their precision and address them in future.